test/gke/resize-cluster.sh

```diff
@@ -28,22 +28,14 @@ script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 
 export KUBECONFIG="${script_dir}/resize-kubeconfig"
 gcloud container clusters get-credentials --project "${project}" --region "europe-west4" management-cluster-0
-
 kubectl get containernodepools.container.cnrm.cloud.google.com "${cluster_name}" -n test-clusters -o yaml | sed "s/nodeCount:.*$/nodeCount: ${node_count}/g" | kubectl replace -f -
 
-gcloud container clusters get-credentials --project "${project}" --region "${region}" "${cluster_uri}"
-
-node_pools=($(gcloud container node-pools list --project "${project}" --region "${region}" --cluster "${cluster_uri}" --format="value(name)"))
-if [ "${#node_pools[@]}" -ne 1 ] ; then
-  echo "expected 1 node pool, found ${#node_pools[@]}"
-  exit 1
-fi
-
 resize_retries=0
 scaled=1
 while [ $resize_retries -lt 4 ]; do
-	echo "trying to scale cluster up"
-	if gcloud container clusters resize --project "${project}" --region "${region}" --node-pool "${node_pools[0]}" --num-nodes "${node_count}" --quiet "${cluster_uri}" ; then
+	echo "waiting for resize operations to finish"
+	operations=$(gcloud container operations list --project "${project}" --filter="status=RUNNING AND operationType=UPDATE_CLUSTER AND targetLink=${cluster_uri}" --format="value(name)")
```

Comment:
afaiu this will get operations list and if there are no running operations on the cluster, it will think the cluster is scaled? I see two issues here:

- the operation hasn't kicked in yet (this is why this whole setup was working in the first place, there was a delay in operation creation from configconnector which will cause this script to report scaling as successful when it has not yet started
- as is, this will check for operation four times in 15s intervals and finish the script. I think that 60s is not enough time for scale operation to finish, so we need to either change `while [ $resize_retries -lt 4 ]; do` to higher retry count or come up with an altogether different way of checking whether cluster is scaled.

Reply:
- From my testing this was instantaneous 🤔 But yea that's a good point.
- I agree, here's a smarter solution that's not just "patching up the existing code".